components/partners/bhaskar/src/main/java/org/mozilla/rocket/bhaskar/Repository.java

```diff
@@ -0,0 +1,182 @@
+package org.mozilla.rocket.bhaskar;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.Observer;
+import android.content.Context;
+import android.support.annotation.Nullable;
+import android.support.v4.util.Pair;
+import android.text.TextUtils;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.mozilla.cachedrequestloader.CachedRequestLoader;
+import org.mozilla.cachedrequestloader.ResponseData;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+
+public class Repository {
+    private int currentPage = FIRST_PAGE;
+    private PageSubscription currentPageSubscription;
+    private static final String SUBSCRIPTION_KEY_NAME = "bhaskar";
+    private static final String DEFAULT_SUBSCRIPTION_URL = "http://appfeed.bhaskar.com/webfeed/apidata/firefox?pageSize=%d&channel_slno=%d&pageNumber=%d";
+    private final String subscriptionUrl;
+    private static final int FIRST_PAGE = 1;
+    private Context context;
+    private int channel;
+    private int pageSize;
+    private String userAgent;
+    private int socketTag;
+    private OnDataChangedListener onDataChangedListener;
```

Comment:
we only need to pass in the listener when we need to. Keeping a reference here may
1. Leak if the implementation is a lifecycle owner.
2. It may be null or not active  at any time